src/python/WMCore/WMSpec/StdSpecs/TaskChain.py

```diff
@@ -657,6 +657,8 @@ def getChainCreateArgs(firstTask=False, generator=False):
             "SizePerEvent": {"type": float, "optional": True, "validate": lambda x: x > 0},
             'PrimaryDataset': {'default': None, 'optional': not generator, 'validate': primdataset,
                                'null': False},
+            "JobExtraMatchRequirements": {"default": "", "type": str,
```

Comment:
This section is in principle for new arguments that are not present in StdBase. Since this argument has already been defined in Stdbase.py below:
 
https://github.com/dmwm/WMCore/pull/12417/files#diff-5a3ac3cd52b6d14d69977560f1fcfe410d45a16f5f683facb1c2ba1fa0a20ef3R1091

Is it really needed in TaskChain here as well?

Reply:
Yes, this definition is required here. As it is set under `getChainCreateArgs()`, so it is enabling this argument under a Task dictionary. While the one in StdBase defines it at the top level of the workflow dictionary (outside any Task/Step).